packages/block-editor/src/components/block-navigation/index.js

```diff
@@ -16,13 +15,41 @@ import { __ } from '@wordpress/i18n';
 import BlockNavigationTree from './tree';
 import { store as blockEditorStore } from '../../store';
 
-function BlockNavigation( {
-	rootBlock,
-	rootBlocks,
-	selectedBlockClientId,
-	selectBlock,
+export default function BlockNavigation( {
+	onSelect = noop,
 	__experimentalFeatures,
 } ) {
+	const { rootBlock, rootBlocks, selectedBlockClientId } = useSelect(
+		( select ) => {
+			const {
+				getBlockHierarchyRootClientId,
+				getSelectedBlockClientId,
+				__unstableGetClientIdsTree,
+				__unstableGetClientIdWithClientIdsTree,
+			} = select( blockEditorStore );
+
+			const _selectedBlockClientId = getSelectedBlockClientId();
+			const _rootBlocks = __unstableGetClientIdsTree();
+			const _rootBlock = _selectedBlockClientId
+				? __unstableGetClientIdWithClientIdsTree(
+						getBlockHierarchyRootClientId( _selectedBlockClientId )
+				  )
+				: null;
+
+			return {
+				rootBlock: _rootBlock,
+				rootBlocks: _rootBlocks,
+				selectedBlockClientId: _selectedBlockClientId,
+			};
+		}
+	);
+	const { selectBlock } = useDispatch( blockEditorStore );
+
+	function selectEditorBlock( clientId ) {
```

Comment:
Any reason to use `function` vs `arrow function`?

Reply:
Not really, no. I don't think there's any practical difference in this case.